Automatically publish your image to ghcr.io (can also add/replace to docker.io!)

Macleykun opened the pull request:
## What's changed?
I've added a workflow that publishes your image everytime you push to main or make a new release (in that case, the tag of that release will be created aswell).

Note, inorder to push to the ghcr.io that's already in your project, you need to allow the CI to push an image to it here:
https://github.com/OpenHealthForAll/open-health/settings/actions
![image](https://github.com/user-attachments/assets/a7b27c70-5914-4b61-9ea3-ca65f5ec4095)

If you want, i can also change the workflow to additional push to docker.io or only to docker.io. Just let me know your preference!
I did get this workflow from GitHub but it's a template i've used a few times before and it works excellent and is very standarized/flexible so you can basically copy paste the entire content and use it in your other projects aswell (given the Container/Dockerfile is in the same place).

Here you can see how the publish went to ghcr.io: https://github.com/Macleykun/open-health/actions/runs/13269126652/job/37044082438
Here you can see the published image: https://github.com/Macleykun/open-health/pkgs/container/open-health

I'll create a seperate PR where i'll optimize your Dockerfile to be more optimized to use caching properly (the layers) but also to run rootless if possible.

I look forward to your response!

Macleykun commented:
Made this into wip until #42 is merged to test the renamed Dockerfile to Containerfile

GwonHyeok commented:
@Macleykun 

We've added new environment variables.

You can see this by checking the .env.example file.

To generate the value for ENCRYPTION_KEY, use the command below
```echo $(head -c 32 /dev/urandom | base64)```

Macleykun commented:
> @Macleykun
> 
> We've added new environment variables.
> 
> You can see this by checking the .env.example file.
> 
> To generate the value for ENCRYPTION_KEY, use the command below `echo $(head -c 32 /dev/urandom | base64)`

i see.. Given this env will be the same because it'll be included in the image, i've added a task in the ci/cd to use the example .env file. It now works, we can also make a new key each time we push but i can imagine that might break stuff with existing installs? Interms of safety we would always be bound to have the encryption env string in the image regardless if we publish the image to pull for users.

Looking forward to your next reply :)

GwonHyeok commented:
I would like to be able to change the ENCRYPTION_KEY at the time of running the container.

Is there a good way to do this?

Macleykun commented:
> I think it would be nice to be able to change the ENCRYPTION_KEY at the time of running the container.
> 
> Is there a good way to do this?

that's possible, so instead of having multiple commands we can put it into a container-entry.sh script and that generates a new encryption_key each time it's run. Is that doable? Or would that break stuff?

GwonHyeok commented:
If ENCRYPTION_KEY is generated every time it is executed, it may cause problems with decrypting existing stored API keys.

I think the default value should be set to .env.example and the user can set the key as a parameter during compose up, so that it can be provided without decryption issues.

Macleykun commented:
> If ENCRYPTION_KEY is generated every time it is executed, it may cause problems with decrypting existing stored API keys.
> 
> I think the default value should be set to .env.example and the user can set the key as a parameter during compose up, so that it can be provided without decryption issues.

i see, i think i'll just cancel this PR, given that we need the key when compiling the node app and can't be changed after it's compiled.
